### ISSUE TITLE
[0.9.5-nightly-b079d20] derivative does not work when data is missing

### ISSUE THREAD

**Issue by parisholley** [influxdb/influxdb#4263]
Building a dash to track my influxdb internal metrics and noticed this.

```
SELECT derivative(sum(points_rx)) / 10 FROM "_internal"."monitor".udp WHERE time > now() - 6h GROUP BY time(10s);
```

if I do not have 6 hours worth of data, it just shows as 

```
name: udp
---------
time			
1443458060000000000	0
```

however moving it to last minute, it works fine

```
name: udp
---------
time			
1443479630000000000	412.80000000004657
1443479640000000000	329
1443479650000000000	392.5
1443479660000000000	428.89999999990687
1443479670000000000	337.20000000006985
```

**Comment by jwilder** [comment]
You need to add `fill(0)` if you have not data.

**Comment by parisholley** [comment]
@jwilder good call, may be a good talking point for #4247

that being said, it seems like a workaround? i could understand seeing no entries for the earlier points, but to not show data for the time that is available, seems unintended?

**Comment by jwilder** [comment]
Fixed by #4292
